Use Kotlin official code style

thomashorta opened the pull request:
Fixes #17704

Remove Kotlin-specific configuration in the `.idea/codeStyles/Project.xml` file in favor of using the default Kotlin official style conventions:

```
<option name="CODE_STYLE_DEFAULTS" value="KOTLIN_OFFICIAL" />
```

Also, reformat all Kotlin files according to this new style to ensure consistency after the above changes.

Most of the changes in this PR is related to the reformatting of .kt files, but the important changes related to the code styles and checks that need revision are in the following files:
- `.idea/codeStylesProject.xml`
- `config/checkstyle.xml` - changes here to make sure the `@Inject` rules target only Java files

**IMPORTANT: after pulling these changes locally I recommend CLOSING Android Studio completely (all windows) and opening it again, as changes in `Project.xml` seem to have some bugs syncing with the IDE.**

_NOTE: this PR is huge, opening the "Files changed" GitHub tab in the browser MIGHT cause the tab to crash or stop responding._

To test:
N/A

## Regression Notes
1. Potential unintended areas of impact
App compilation errors, linting errors, or runtime errors.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Limited manual testing (change affects _only_ formatting, should not introduce behavior changes)

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [X] I have completed the Regression Notes.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Commit:
keep relevant Kotlin configs in Project.xml and set defaults

Commit:
remove XML section from Projects.xml to match recent AS configs

Commit:
format all Kotlin files with default rules

Commit:
update some checkstyle.xml rules to target only java

Talking with Alex Forcier, who made the original checkstyle rules in the
project, we agreed that the @Inject rules might not make a whole lot of
sense based on Kotlin code style defaults (which are set to break the line
after each annotation), so I changed the checkstyle rules related to that
to only target `Java` files.

Commit:
move the checkstyle ignore comment to the correct line

It was moved during the Kotlin file refactors by mistake, so I'm moving it back

ravishanker reviewed (approved):
Thanks for undertaking this 👍

Commit:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into issue/17704-code-style-kotlin

Commit:
reformat all files after merging trunk into this branch

One thing I noticed is that empty lines were added between annotated properties
this time. After some investigation, it seems Android Studio is unable to do all
formatting at once so before, when I ran reformat once, it just added a line
break after each annotation, and this time it added blank lines between
annotated properties, which is also part of the convention.

With that in mind, I ran reformatting twice, so all changes would be covered on
top of the most recent changes in trunk.

thomashorta commented:
I just synced this branch with `trunk` and reformatted everything again.

One thing I noticed is that empty lines were added between annotated properties this time. After some investigation, it seems Android Studio is unable to do all formatting at once so before when I ran reformat once, it just added a line break after each annotation, and this time it added blank lines between annotated properties, which is also part of the convention.

With that in mind, I ran reformatting twice, so formatting is properly applied on top of the most recent changes in `trunk`.

---

### Practical example

#### Before
```kotlin
@Inject lateinit var dispatcher: Dispatcher
@Inject lateinit var accountStore: AccountStore
@Inject lateinit var siteStore: SiteStore
```

#### After the first reformat (before today's changes)
```kotlin
@Inject
lateinit var dispatcher: Dispatcher
@Inject
lateinit var accountStore: AccountStore
@Inject
lateinit var siteStore: SiteStore
```

#### After the second reformat (after today's changes)
```kotlin
@Inject
lateinit var dispatcher: Dispatcher

@Inject
lateinit var accountStore: AccountStore

@Inject
lateinit var siteStore: SiteStore
```

mokagio reviewed (approved):
I just wanted to endorse the idea of using the official style guide, in particular for an open source project such as this WordPress Android. I feel like it will remove the need to discuss personal style preferences among existing contributors and help new ones get started.

Commit:
remove empty file created during merge conflict resolution

irfano commented on .idea/codeStyles/Project.xml:
```diff
@@ -238,20 +224,7 @@
-      <option name="METHOD_ANNOTATION_WRAP" value="1" />
-      <option name="FIELD_ANNOTATION_WRAP" value="1" />
-      <option name="ENUM_CONSTANTS_WRAP" value="5" />
+      <option name="CODE_STYLE_DEFAULTS" value="KOTLIN_OFFICIAL" />
```
Do we have to specify `CODE_STYLE_DEFAULTS` as `KOTLIN_OFFICIAL`? If we don't define `CODE_STYLE_DEFAULTS`, isn't the default value `KOTLIN_OFFICIAL` already?

thomashorta replied:
That was my first thought as well, but I found that if we don't have the CODE_STYLE_DEFAULTS set explicitly to `KOTLIN_OFFICIAL` then AS uses its own defaults (which seem to match "Kotlin obsolete IntelliJ IDEA codestyle" from the CodeStyle settings).

![image](https://user-images.githubusercontent.com/5091503/210785871-b4d636f8-03aa-48a1-a4db-335b491b285f.png)

See this video I made checking the `Project.xml` on a new project and changing codestyle settings. _Note: for better immersion while watching the video, imagine myself shrugging in disbelief at the end of it!_

https://user-images.githubusercontent.com/5091503/210786489-92d56cd5-3f54-45e4-954f-ce898cb3d8a3.mov

irfano replied:
I understand. I assumed Android Studio would use the most up-to-date style by default. Apparently, it uses an obsolete codestyle instead. Thanks for the clarification.

irfano reviewed (approved):
Thank you very much for dealing with this. ❤️ 
I'd say we don't need to be afraid of the vast amount of changed files since they are only style changes. Before, I only had issues with reformatting when I did it for XML files. Sometimes it changes the order of elements, and that causes changes in the layout. But we're not updating XML files here, so it looks safe. 👍🏻

Commit:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into issue/17704-code-style-kotlin

Commit:
reformat again after syncing with trunk

thomashorta commented:
I know this has a few approvals already but I will wait for @ParaskP7 to take a look as well as he did raise some codestyle discussions in the past and had some points as well when I proposed this change. I believe he will be able to take a look next week.

Until then I will keep syncing this PR with `trunk` daily so it doesn't get too behind in terms of changes and conflicts.

ParaskP7 commented:
👋 @thomashorta and happy new year! 🎉 

> I know this has a few approvals already but I will wait for @ParaskP7 to take a look as well as he did raise some codestyle discussions in the past and had some points as well when I proposed this change.

Hey, please don't be blocked by me on that, my code style related points were mainly on the version control side of things and the `.idea` files. This change is trying to deal with a different problem and I am all-in on it no matter, so this is a big 🟢 from me, thank you! 🥇 

> I believe he will be able to take a look next week.

I did take a quick look, just to unblock you so you could merge this PR as soon as possible. I mainly looked at the below commits and some random file changes just to make sure that the indentation is the expected one:
- [keep relevant Kotlin configs in Project.xml and set defaults](https://github.com/wordpress-mobile/WordPress-Android/commit/883398f892c933fc002d07cedb126a1803176e59)
- [remove XML section from Projects.xml to match recent AS configs](https://github.com/wordpress-mobile/WordPress-Android/commit/84fcdfefed82c443fb86457e7103c3086c3bdf95)
- [update some checkstyle.xml rules to target only java](https://github.com/wordpress-mobile/WordPress-Android/commit/87c5892071b1d5e930407cf067539c3380ca1980)

Everything looks good to me, many thanks for this update! 💯 

> Until then I will keep syncing this PR with trunk daily so it doesn't get too behind in terms of changes and conflicts.

I hope this is the last time you sync this PR with `trunk`, let's do this! 🤞 😅 🟢

Commit:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into issue/17704-code-style-kotlin

Accepts "theirs" changes fully, to get exactly what trunk has.

Commit:
reformat again after syncing with trunk
